api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -66,6 +77,16 @@ static Schema schema() {
    */
   int partitionSpecId();
 
+  /**
+   * @return the sequence number of the commit that added the manifest file
+   */
+  long sequenceNumber();
+
+  /**
+   * @return the lowest sequence number of any data file in the manifest
+   */
+  long minSequenceNumber();
```

Comment:
Why need to define this min sequence number ?  for manifest file filtering purpose in read path ?

Reply:
This is additional information for job planning and metadata maintenance.

This may change, but the initial idea is that if we have the min sequence number of a manifest, we can use that to filter other manifests for planning. For example, if I have a manifest with delete files that was written at sequence number 14, but only manifests with data files written at sequence number 16 and later, I can ignore the manifest with delete files.

We can do something similar when maintaining metadata. If we detect that a delete file is older than all data files then we can remove it.